src/eventarc/eventarc.ts

```diff
@@ -0,0 +1,287 @@
+/*!
+ * Copyright 2020 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { App } from '../app';
+import * as validator from '../utils/validator';
+import { FirebaseEventarcError, toCloudEventProtoFormat } from './eventarc-utils';
+import {
+  HttpRequestConfig, HttpClient, HttpError, AuthorizedHttpClient
+} from '../utils/api-request';
+import { FirebaseApp } from '../app/firebase-app';
+import * as utils from '../utils';
+import { PrefixedFirebaseError } from '../utils/error';
+import { CloudEvent } from './cloudevent';
+
+const EVENTARC_API = 'https://eventarcpublishing.googleapis.com/v1';
+const FIREBASE_VERSION_HEADER = {
+  'X-Firebase-Client': `fire-admin-node/${utils.getSdkVersion()}`,
+};
+const CHANNEL_NAME_REGEX = /^(projects\/([^/]+)\/)?locations\/([^/]+)\/channels\/([^/]+)$/;
+const DEFAULT_CHANNEL_REGION = 'us-central1';
+
+/**
+ * Channel options interface.
+ */
+export interface ChannelOptions {
+  /**
+   * An array of allowed event types. If specified, publishing events of
+   * unknown types will be a no op. When not provided, no even filtering is
+   * performed.
+   */
+  allowedEventTypes?: string[] | string | undefined
+}
+
+/**
+ * Eventarc service bound to the provided app.
+ */
+export class Eventarc {
+
+  private readonly appInternal: App;
+
+  /**
+   * @internal
+   */
+  constructor(app: App) {
+    if (!validator.isNonNullObject(app) || !('options' in app)) {
+      throw new FirebaseEventarcError(
+        'invalid-argument',
+        'First argument passed to Eventarc() must be a valid Firebase app instance.',
+      );
+    }
+
+    this.appInternal = app;
+  }
+
+  /**
+   * The {@link firebase-admin.app#App} associated with the current `Eventarc` service
+   * instance.
+   *
+   * @example
+   * ```javascript
+   * var app = eventarc.app;
+   * ```
+   */
+  get app(): App {
+    return this.appInternal;
+  }
+  
+  /**
+   * Creates a reference to Eventarc default Firebase channel which can then
+   * be used to publish events:
+   * `projects/{project}/locations/us-central1/channels/firebase`
+   * 
+   * @param options - (optional) additional channel options
+   * @returns Eventarc channel reference for publishing events.
+   */
+  public channel(options?: ChannelOptions): Channel;
+  
+
+  /**
+   * Creates a reference to Eventarc channel using provided channel resource name.
```

Comment:
Suggest "the Eventarc default . . . using the provided . . ."

Reply:
Done